rc/filetype/graphql.kak

```diff
@@ -27,22 +27,46 @@ hook -group graphql-highlight global WinSetOption filetype=graphql %{
 }
 
 
-provide-module graphql %(
+provide-module graphql %§ 
 
 # Highlighters
 # ‾‾‾‾‾‾‾‾‾‾‾‾
 
 add-highlighter shared/graphql regions
 add-highlighter shared/graphql/code default-region group
-add-highlighter shared/graphql/string region '"' (?<!\\)(\\\\)*" fill string
-add-highlighter shared/graphql/comment region '#' \n fill comment
+add-highlighter shared/graphql/line-description region '#' '\n' fill comment
+add-highlighter shared/graphql/block-description region '"""' '"""' fill comment
+add-highlighter shared/graphql/description region '"' '"\s*\n' fill comment
+add-highlighter shared/graphql/object region -recurse \{ [{] [}] regions
 
-add-highlighter shared/graphql/code/ regex \b(fragment|query|mutation|on)\b 0:keyword
-add-highlighter shared/graphql/code/ regex \b(true|false|null|\d+(?:\.\d+)?(?:[eE][+-]?\d*)?)\b 0:value
-add-highlighter shared/graphql/code/ regex \$[a-zA-Z0-9]+\b 0:value
-add-highlighter shared/graphql/code/ regex @(?:include|skip) 0:meta
-add-highlighter shared/graphql/code/ regex \b(Boolean|Float|ID|Int|String)\b 0:type
-add-highlighter shared/graphql/code/ regex ! 0:operator
+# Objects
+add-highlighter shared/graphql/object/line-description region '#' '\n' fill comment
+add-highlighter shared/graphql/object/block-description region '"""' '"""' fill comment
+add-highlighter shared/graphql/object/field default-region group
+add-highlighter shared/graphql/object/field/ regex ([A-Za-z][A-Za-z0-9_-]*)(?:\([^)]*\))?\h*[:{] 1:attribute
+add-highlighter shared/graphql/object/field/ regex ^\h*([A-Za-z][A-Za-z0-9_-]*)\h*$ 1:attribute
+
+# Values
+add-highlighter shared/graphql/object/field/values regex \b(true|false|null|\d+(?:\.\d+)?(?:[eE][+-]?\d*)?)\b 0:value
+add-highlighter shared/graphql/object/field/variables regex \$[a-zA-Z0-9]+\b 0:variable
+add-highlighter shared/graphql/object/field/string regex '"[^"]*"' 0:string
```

Comment:
This seems like it would not catch escaped quotes as before.  Maybe something like `'"([^"]|\\.)*"'`?

Reply:
good catch! fixed